tests/test_cuboid.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+
+import pytest
+import unittest
+from ddt import ddt, data, unpack
+
+import numpy as np
+from broni.shapes.primitives import Cuboid
+
+
+@ddt
+class TestCuboid(unittest.TestCase):
+    def setUp(self):
+        pass
+
+    def tearDown(self):
+        pass
+
+    def test_invalid(self):
+        with pytest.raises(ValueError) as e:
+            assert Cuboid(0, 0, 0, 0, 0, 0)
+
+    @data(
+        (np.array([[1, 1, 1]]), Cuboid(0, 0, 0, 10, 10, 10),
+         np.array([True])),  # one point
+        (np.array([[0, 0, 0]]), Cuboid(0, 0, 0, 10, 10, 10),
+         np.array([True])),  # edge point
+        (np.array([[-1, 0, 0]]), Cuboid(0, 0, 0, 10, 10, 10),
+         np.array([False])),  # outside
+        (np.array([[-1, 0, 0], [0, 0, 0], [1, 1, 1], [11, 11, 11]]), Cuboid(0, 0, 0, 10, 10, 10),
+         np.array([False, True, True, False], dtype=bool)),  # multiple points
+        (np.array([[-1, 0, 0]]), Cuboid(-10, -10, -10, 0, 0, 0),
+         np.array([True])),  # a cuboid in another quadrant
+        (np.array([]), Cuboid(0, 0, 0, 1, 1, 1),
+         np.array([])),  # empty trajectory
+    )
+    @unpack
+    def test_intersections(self, trajectory, shape, expected):
+        np.testing.assert_array_equal(shape.intersect(trajectory), expected)
+
+    @data(
+        (Cuboid(0, 0, 0, 10, 10, 10),
+         np.array([[[0., 10., 10., 0., 0.],
+                    [0., 10., 10., 0., 0.],
+                    [0., 10., 10., 0., 0.],
+                    [0., 10., 10., 0., 0.]],
+
+                   [[0., 0., 10., 10., 0.],
+                    [0., 0., 10., 10., 0.],
+                    [0., 0., 0., 0., 0.],
+                    [10., 10., 10., 10., 10.]],
+
+                   [[0., 0., 0., 0., 0.],
+                    [10., 10., 10., 10., 10.],
+                    [0., 0., 10., 10., 0.],
+                    [0., 0., 10., 10., 0.]]]),
+         )
+    )
+    @unpack
+    def test_to_mesh(self, shape, expected):
```

Comment:
This one may be difficult to write, instead of checking each vertex, maybe some properties like center and volume could be checked.

Reply:
Most likely temporary.